src/dataToEsm.js

```diff
@@ -2,32 +2,24 @@ import makeLegalIdentifier from './makeLegalIdentifier';
 import tosource from 'tosource';
 
 // convert data object into separate named exports (and default)
-export default function dataToNamedExports ( obj, compact ) {
-  const _ = compact ? '' : ' ';
-  const nl = compact ? '' : '\n';
-  let output = '';
-  let defaultExports = compact ? '' : nl;
-  const usedLegalKeys = Object.create( null );
-  let index = 0;
-  const keys = Object.keys( obj );
-  for ( let i = 0; i < keys.length; i++ ) {
-    const key = keys[i];
-    const legalKey = makeLegalIdentifier( key );
-    let serialized = tosource( obj[key], null, compact ? false : '  ' );
-    output += `export const ${ legalKey }${ _ }=${ _ }${ serialized };${ nl }`;
-    while ( usedLegalKeys[legalKey + (index || '')] )
-      index++;
-    index = 0;
-    usedLegalKeys[legalKey + ( index || '' )] = true;
-    if ( defaultExports.length > 1 )
-      defaultExports += `,${ nl }`;
-    if ( key === legalKey )
-      defaultExports += `${ _ }${ _ }${ key }`;
-    else
-      defaultExports += `${ _ }${ _ }${ `'${ key }'` }:${ _ }${ legalKey }`;
-  }
-  if ( defaultExports.length > 3 )
-    defaultExports += nl;
-  output += `export default${_}{${ defaultExports }};`;
-  return output;
+export default function dataToNamedExports ( data, options = {} ) {
+  const t = options.compact ? '' : 'indent' in options ? options.indent : '\t';
+  const _ = options.compact ? '' : ' ';
+  const n = options.compact ? '' : '\n';
+  const declarationType = options.preferConst ? 'const' : 'var';
+
+  let namedExportCode = '';
+  const defaultExportRows = [];
+  Object.keys( data ).forEach(key => {
```

Comment:
There's a bit of an issue with that as Buble doesn't support transforming for-of. I think we're ok when not in tight loops for this sort of optimization though.

Reply:
I was actually suggesting to use the exact code as used in the initial version—a classic `for` loop with an iterator variable. It's still the fastest way to iterate. `for...of` loops in TypeScript are only fast because due to our compilation target, it is transpiled to a classic `for` loop. Real `for...of` loops are actually quite a bit slower due to the hidden iterator evaluations anyway.

Reply:
Ah I completely forgot I did that. Sure.